src/Components/Web.Extensions/src/InputFile/InputFileChangeEventArgs.cs

```diff
@@ -11,18 +11,63 @@ namespace Microsoft.AspNetCore.Components.Web.Extensions
     /// </summary>
     public class InputFileChangeEventArgs : EventArgs
     {
+        private readonly IReadOnlyList<IBrowserFile> _files;
+        private readonly long _maxFileSize;
+        private readonly int _maxAllowedFiles;
+
+        private bool _validated;
+
         /// <summary>
-        /// The updated file entries list.
+        /// Gets the updated file entries list.
         /// </summary>
-        public IReadOnlyList<IBrowserFile> Files { get; }
+        public IReadOnlyList<IBrowserFile> Files
+        {
+            get
+            {
+                ValidateFiles();
```

Comment:
Could we do this in `InputFile.NotifyChange`? That way the component is responsible for the code and this is a POCO type

Reply:
How would the user catch this exception? `NotifyChange` is initiated from a JS callback.